### PR TITLE
Use QLoggingCategories for logging

### PR DESCRIPTION
**Pull request by pvuorela**
More standard logging, allows better control on what gets printed and avoids collision with QT_MESSAGE_PATTERN, etc. Also console.log() etc will not be filtered out by default which to me has appeared to confuse some people trying to hack the keyboard before finding out the environment variable.

### CODE REVIEW

File: passthroughserver/main.cpp [hunk 1]
```diff
@@ -23,64 +23,33 @@
...
+        defaultLoggingFilter(category);
     }
-
-    return debugEnabled == 1;
 }
 
-void outputMessagesToStdErr(QtMsgType type,
-                            const QMessageLogContext &context,
-                            const QString &msg)
+void disableMInputContextPlugin()
```
Comment (dobey) [inline passthroughserver/main.cpp:48]: It seems this wasn't changed, but was removed and then added back in a different location, which makes the diff to `main.cpp` here quite a bit weird and extraneous. Could you perhaps move it back, and rebase so that the diff is cleaner and easier to understand without this unrelated change? Thanks.

File: passthroughserver/main.cpp [hunk 2]
```diff
@@ -23,64 +23,33 @@
...
-    setenv("QT_IM_MODULE", "none", true);
-}
-
-bool isDebugEnabled()
+void loggingCategoryFilter(QLoggingCategory *category)
 {
-    static int debugEnabled = -1;
-
-    if (debugEnabled == -1) {
+    if (qstrcmp(category->categoryName(), "driver.usb") == 0) {
```
Comment (dobey) [inline passthroughserver/main.cpp:36]: Why is this necessary? And why would we only check for `MALIIT_DEBUG` if `driver.usb` is the category? I don't understand this. If this is necessary, please explain it with a comment in the code. Thanks.

**Review by dobey — CHANGES_REQUESTED** [review]

File: passthroughserver/main.cpp (hunk 2 above)
Comment (pvuorela) [reply passthroughserver/main.cpp]: Duh, used a snippet from the docs and somehow ended up having the actual category name in a git stash. Fixed now.

File: passthroughserver/main.cpp (hunk 1 above)
Comment (pvuorela) [reply passthroughserver/main.cpp]: Actually I didn't touch it, but git considered the old log filtering to be similar enough to the new category filtering so it ended up showing delta with this moved instead. But since the function here is just wrapping one function call, let's just get rid of he extra indirection.

File: passthroughserver/main.cpp [hunk 3]
```diff
@@ -23,63 +23,25 @@
...
-    case QtWarningMsg:
-        fprintf(stderr, "WARNING: %s\n", raw);
-        break;
-    case QtCriticalMsg:
-        fprintf(stderr, "CRITICAL: %s\n", raw);
-        break;
-    case QtFatalMsg:
-        fprintf(stderr, "FATAL: %s\n", raw);
-        abort();
+        category->setEnabled(QtDebugMsg, enabled);
```
Comment (dobey) [inline passthroughserver/main.cpp:42]: As the default level is Warning as set by the declaration of the category, can't we drop the extra `bool enabled` usage here, and just always all this with true if the `debugEnvVar` check is true?

**Review by dobey — CHANGES_REQUESTED** [review]

File: passthroughserver/main.cpp (hunk 3 above)
Comment (pvuorela) [reply passthroughserver/main.cpp]: Just doing that wouldn't work because by default all logging is turned on and the default filter would be setting it off here. 

But instead moved now the default handler always done first and the debug override happening later and like suggested to only set debug true. Makes QT_LOGGING_RULES honored better.

**Review by dobey — APPROVED** [review]
Looks OK now. Thanks.
